src/main/java/seedu/weme/model/util/SampleDataUtil.java

```diff
@@ -15,26 +21,51 @@
  * Contains utility methods for populating {@code MemeBook} with sample data.
  */
 public class SampleDataUtil {
-    public static Meme[] getSampleMemes() {
-        return new Meme[] {
-            new Meme(new ImagePath("src/main/resources/memes/charmander_meme.jpg"),
-                new Description("A meme about Char and charmander."),
-                getTagSet("charmander")),
-            new Meme(new ImagePath("src/main/resources/memes/doge_meme.jpg"),
-                    new Description("A meme about doge."),
-                    getTagSet("doge")),
-            new Meme(new ImagePath("src/main/resources/memes/joker_meme.jpg"),
-                    new Description("A meme about joker."),
-                    getTagSet("joker")),
-            new Meme(new ImagePath("src/main/resources/memes/toy_meme.jpg"),
-                    new Description("A meme about toy."),
-                    getTagSet("toy"))
+    public static Meme[] getSampleMemes(ReadOnlyUserPrefs userPrefs) {
+        // array of sample memes from resources folder
+        MemeFieldsContainer[] memeFields = new MemeFieldsContainer[]{
+            new MemeFieldsContainer("memes/5E88E068898624CAA37E4DFA50E1E240470EB2F8.png",
+                    "A meme about doge.", "doge"), // doge
+            new MemeFieldsContainer("memes/F32C68AD80EE2754BCD012A37530A16239E6A587.jpg",
+                    "A meme about Char and charmander.", "charmander"), // charmander
+            new MemeFieldsContainer("memes/EB469F0CC92EFE4589C7529CD0AA2D2E49DE96FB.png",
+                    "A meme about joker.", "joker"), // joker
+            new MemeFieldsContainer("memes/FA03CFED3E876472A3EF9745EC9080879DF417EC.png",
+                    "A meme about toy.", "toy", "jokes"), // toy
+            new MemeFieldsContainer("memes/C37F673DE2C942D0330F02AC62598EC7D4736723.jpg",
+                    "A meme about a test.", "test") // test
         };
+        return createSampleMemes(memeFields, userPrefs);
+    }
+
+    /**
+     * Copies meme images from Resource folder to the Data folder.
+     * @param memeFields the data for the memes in the resource folder
+     * @param userPrefs the user preferences for this instance of weme
+     * @return an array of Memes to import
+     */
+    public static Meme[] createSampleMemes(MemeFieldsContainer[] memeFields, ReadOnlyUserPrefs userPrefs) {
+        ClassLoader classLoader = SampleDataUtil.class.getClassLoader();
+        Meme[] copiedMemes = new Meme[memeFields.length];
+
+        for (int i = 0; i < copiedMemes.length; i++) {
+            String path = memeFields[i].getImagePath();
+            String newPath = userPrefs.getMemeImagePath().toString() + "/" + FileUtil.getFileName(path);
+            try {
+                FileUtil.copy(classLoader.getResourceAsStream(path), Paths.get(newPath));
+            } catch (IOException e) {
```

Comment:
Should we catch and swallow `FileAlreadyExistsException` in case the meme image folder isn't empty?